vector/src/main/java/im/vector/app/features/home/room/detail/TimelineFragment.kt

```diff
@@ -1974,6 +2004,32 @@ class TimelineFragment :
         }
     }
 
+    /**
+     * Navigate to pinned messages for the current room using the PinnedMessagesActivity.
+     */
+    private fun navigateToPinnedMessages() = withState(timelineViewModel) { state ->
+        val pinnedEventId = timelineViewModel.getIdOfLastPinnedEvent()
```

Comment:
I do not see the point of passing `pinnedEventId` to `PinnedMessagesTimelineArgs`. Same for other parameters. In the PinnedEvent Activity, the ViewModel should observe the state event with pinned event and get the Event from that list.

The only needed parameter is the `roomId`.

Do I miss something?